addons/website_slides/static/src/js/slides_course_quiz.js

```diff
@@ -267,6 +346,248 @@ odoo.define('website_slides.quiz', function (require) {
+            } else {
+                $elem.replaceWith(QWeb.render('slide.quiz.new.question.button'));
+            }
+        },
+
+        _displayUpdatedQuestion: function (response) {
+            response.data.$elem.html(QWeb.render('slide.quiz.question.view', { question:response.data }));
+        },
+
+        _resetDisplay: function (reset) {
+            if (reset.data.update) {
+                reset.data.$elem.html(reset.data.$oldElem);
+            } else {
+                var $elem = this.$('.o_wslides_js_lesson_quiz_new_question');
+                if (this.quiz.questions.length > 0) {
+                    $elem.html(QWeb.render('slide.quiz.new.question.button'));
+                } else {
+                    $elem.html(QWeb.render('slide.quiz.new.quiz.button'));
+                }
+            }
+        },
+
+        _deleteQuestion: function(event) {
+            var question = event.data;
+            question.remove();
+            this.quiz.questions = this.quiz.questions.filter(obj => { return obj.id != question[0].dataset.questionId });
+            this._reorderQuestions();
+            var $newQuestion = this.$('.o_wslides_js_lesson_quiz_new_question .o_wslides_quiz_question_sequence');
+            $newQuestion.text(Number($newQuestion.text()) - 1);
+        },
+    });
+
+    var ConfirmationDialog = Dialog.extend({
+        template: 'slide.quiz.confirm.deletion',
+        xmlDependencies: Dialog.prototype.xmlDependencies.concat(
+            ['/website_slides/static/src/xml/slide_quiz_create.xml']
+        ),
+
+        init: function (parent, options, question) {
+            options = _.defaults(options || {}, {
+                title: _t('Delete Question'),
+                buttons: [
+                    { text: _t('Yes'), classes: 'btn-primary', click: this._confirm },
+                    { text: _t('No'), close: true}
+                ],
+                size: 'medium'
+            });
+            this._super(parent, options);
+            this.question = question;
+        },
+
+        _confirm: function () {
+            var self = this;
+            this._rpc({
+                route: '/slides/slide/quiz/question_delete',
+                params: {
+                    id: this.question.data('question-id')
+                }
+            }).then(function () {
+                self.trigger_up('deleteQuestion', self.question);
+                self.close();
+            }, function (error) {
+                console.log(error);
+            });
+        }
+    });
+
+    /**
+     * This Widget is responsible of displaying the question inputs when adding a new question or when updating an
+     * existing one. When validating the question it makes an RPC call to the server and trigger an event for
+     * displaying the question by the Quiz widget.
+     */
+    var QuestionWidget = publicWidget.Widget.extend({
```

Comment:
The name is confusing since it's only used when creating or updating questions.
Maybe `QuestionFormWidget` ?

Reply:
Also, as it's like 100+ lines of code, this widget should be in its own file.